kobo/hub/models.py

```diff
@@ -252,7 +252,7 @@ def _get_version(self):
 
     def running_tasks(self):
         """Return list of running tasks on this worker."""
-        return Task.objects.running().filter(worker=self)
+        return Task.objects.opened().filter(worker=self)
```

Comment:
The term "running" doesn't match any exact state in the DB, so it seems subject to interpretation.  I don't doubt that this change fixes bugs but I really don't like the fact that this has now set up *two* different interpretations of the term "running" in the same code base:

- in `Worker.running_tasks`, "running" means: tasks in `OPEN` state
- in `TaskManager.running`, "running" means: tasks in `ASSIGNED` or `OPEN` state

They can't both be right.  Is there some way to resolve this? (without renaming/removing methods or other major backwards-incompatible things)

Or if not, at least to have a comment here or expand the doc string here to justify it?  Taken out of context, this change looks more like a bug is being introduced rather than fixed.  This definitely risks someone else seeing this code later and thinking: there's a Task.objects.running() method but it's not called here, that must be a bug, I'll "fix" it.

Reply:
I understand you concerns.  There are two other options how to implement this without creating this inconsistency:

1. Edit `hub.get_worker_tasks` to filter out all `ASSIGNED` because this XML-RPC call is always used by the worker in a context where only tasks with an `OPEN` state are expected.
2. Filter all `ASSIGNED` tasks from the list returned by `hub.get_worker_tasks` on the worker.

I've rebased this PR with the implementation of the first option but I'll rework it if you prefer the second one or have other ideas.